.github/workflows/docker.yml

```diff
@@ -2,6 +2,7 @@ name: Build and push Docker images to Docker Hub
 
 on:
   workflow_dispatch:
+  pull_request:
```

Comment:
Don't forget to revert this, we don't want every PR pushed to docker right?

Reply:
Yes, it's just temporarily changed to test on this PR

Reply:
Yes just testing if the image builds in this PR

Reply:
I think the build completed (and it works locally for me), just the login failed before you made the other changes to the file.

Reply:
The image builds locally for me as well, the problem seems to be in the CI command.

Reply:
Due to the force push I can't easily get back to it, but to me it looked like the CI did finish before, it just failed to push to dockerhub

Reply:
Have a look [here](https://github.com/pybamm-team/PyBaMM/actions/runs/10285168300/job/28462799902?pr=3901), these are the logs from the most recent push.

Reply:
It fails on build, can u try with simple `docker build -t pybamm:latest -f scripts/Dockerfile .` once

Reply:
Seems to be working now 😃

Reply:
Nah you're good, it's a little confusing as `docker buildx` requires login for build as well. I missed it before.

Reply:
Ok revert the workflow and we can get it approved and squashed